.github/workflows/ci_linux.yml

```
@@ -0,0 +1,120 @@
+          sudo apt-get install -y build-essential curl python3-pip git uncrustify valgrind
+          python3 -m pip install --upgrade pip
+          python3 -m pip install gcovr
+
+          # CMake 3.15.7 (to mirror Jenkins build environment)
+          CMAKE_OLD=3.15.7
+          curl -L "https://github.com/Kitware/CMake/releases/download/v${CMAKE_OLD}/cmake-${CMAKE_OLD}-Linux-x86_64.tar.gz" -o cmake-old.tgz
+          sudo tar -C /opt -xzf cmake-old.tgz
+          echo "/opt/cmake-${CMAKE_OLD}-Linux-x86_64/bin" >> "$GITHUB_PATH"
+
+          # Newer CTest for JUnit export
+          CMAKE_NEW=3.27.9
+          curl -L "https://github.com/Kitware/CMake/releases/download/v${CMAKE_NEW}/cmake-${CMAKE_NEW}-linux-x86_64.tar.gz" -o cmake-new.tgz
+          sudo tar -C /opt -xzf cmake-new.tgz
+          echo "CTEST_NEW=/opt/cmake-${CMAKE_NEW}-linux-x86_64/bin/ctest" >> "$GITHUB_ENV"
+
+      - name: Show cmake version
+        run: cmake --version
+
+      - name: Configure
+        run: |
+          mkdir -p build
+          cd build
+          cmake ../ci/linux -DCMAKE_BUILD_TYPE=Release
+
+      - name: Build
+        run: |
+          cd build
+          make -j"$(nproc)"
+
+      # --- Run tests with newer CTest to produce JUnit ---
+      - name: Run tests and export JUnit
+        shell: bash
+        run: |
+          "${CTEST_NEW}" --version
+          "${CTEST_NEW}" --test-dir build/microcdr-build \
+            --output-on-failure \
+            --no-tests=error \
+            --output-junit "$GITHUB_WORKSPACE/build/CTestResults.xml"
+
+      - name: Generate coverage (gcovr Cobertura XML + HTML)
+        shell: bash
+        run: |
+          EXCLUDES=(
+            '--exclude-unreachable-branches'
+            '--exclude' 'ci/'
+            '--exclude' 'test/'
+            '--exclude' 'build/'
+          )
+          gcovr -x -r . "${EXCLUDES[@]}" -o build/coverage.xml
+          gcovr --html-details -r . "${EXCLUDES[@]}" -o build/coverage.html
+
+      - name: Uncrustify check
+        shell: bash
+        run: |
+          curl -L https://raw.githubusercontent.com/eProsima/cpp-style/master/uncrustify.cfg -o uncrustify.cfg
+          set +e
+          find src test include examples \( -name "*.cpp" -o -name "*.c" -o -name "*.h" -o -name "*.hpp" \) \
+            -exec uncrustify -c uncrustify.cfg --check {} +
+          UNCR_EXIT=$?
+          set -e
+          if [ $UNCR_EXIT -ne 0 ]; then
+            echo "Code style violations found by uncrustify."
+            exit $UNCR_EXIT
+          fi
+
+      - name: Upload artifacts (CTest & Coverage)
+        uses: actions/upload-artifact@v4
+        with:
+          name: test-and-coverage
+          path: |
+            build/CTestResults.xml
+            build/coverage.xml
+            build/coverage.html
+          if-no-files-found: warn
+
+      - name: Publish unit test results (JUnit)
+        uses: EnricoMi/publish-unit-test-result-action@v2
+        if: always()
```

Perhaps we could skip this if something was wrong before:


```suggestion
      - name: Publish unit test results (JUnit)
        uses: EnricoMi/publish-unit-test-result-action@v2
        if: success()
```

---

We want to publish test results if some of the tests fail.
Switched to `if: (!cancelled())`